adal/src/androidTest/java/com/microsoft/aad/adal/HttpDialogTests.java

```diff
@@ -24,29 +24,33 @@
 package com.microsoft.aad.adal;
 
 import android.annotation.SuppressLint;
+import android.content.Context;
 import android.content.pm.PackageInfo;
 import android.content.pm.PackageManager;
 import android.content.pm.Signature;
-import android.test.AndroidTestCase;
+import android.support.test.InstrumentationRegistry;
 import android.util.Base64;
 import android.util.Log;
 
+import org.junit.Before;
+
 import java.security.MessageDigest;
 
-public class HttpDialogTests extends AndroidTestCase {
+public class HttpDialogTests {
 
     private static final String TAG = "HttpDialogTests";
 
     private byte[] mTestSignature;
 
     private String mTestTag;
 
-    @Override
+    private Context mContext;
+
     @SuppressLint("PackageManagerGetSignatures")
+    @Before
     protected void setUp() throws Exception {
-        super.setUp();
-        getContext().getCacheDir();
-        System.setProperty("dexmaker.dexcache", getContext().getCacheDir().getPath());
+        mContext = InstrumentationRegistry.getContext();
+        System.setProperty("dexmaker.dexcache", mContext.getCacheDir().getPath());
```

Comment:
This is just making necessary changes after deprecating the AndroidTestCase. I saw code updating this settings several times in other classes that were written previously. I didn't change the logic. It seems when it's needed, update the settings in advance, and it works.